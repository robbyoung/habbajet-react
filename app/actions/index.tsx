export * from './actionTypes';
export * from './addHabbajet';
export * from './addHabitResult';
export * from './updateBudget';
export * from './resetHabbajet';
export * from './loadState';
export * from './selectHabbajet';
export * from './updateEditorField';
export * from './validateEditor';
export * from './clearEditor';
