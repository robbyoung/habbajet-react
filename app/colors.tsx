export const white = '#ffffff';
export const lightGrey = 'F5F5F5';
export const grey = '#959595';

export const habbajetColors = [
    '#8066C2', // Purple
    '#2AAD5E', // Green
    '#E04747', // Red
    '#7F90EC', // Blue
    '#B1B952', // Gold
];

export const complementColors: {[id: string]: string} = {
    '#8066C2': '#C2A066',
    '#2AAD5E': '#AD372A',
    '#E04747': '#47E047',
    '#7F90EC': '#ECB27F',
    '#B1B952': '#B952B0',
};
