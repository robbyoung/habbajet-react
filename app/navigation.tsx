import {Navigation} from 'react-native-navigation';
import React from 'react';
import HabbajetScreen from './containers/habbajetScreen';
import LoadingScreen from './containers/loadingScreen';
import {Provider} from 'react-redux';
import store from './store';

enum Screens {
    Loading = 'Loading',
    Habbajet = 'Habbajet',
}

export const STACK_NAVIGATOR = 'StackNavigator';

Navigation.registerComponent(
    Screens.Loading,
    () => () => (
        <Provider store={store}>
            <LoadingScreen />
        </Provider>
    ),
    () => LoadingScreen,
);

Navigation.registerComponent(
    Screens.Habbajet,
    () => () => (
        <Provider store={store}>
            <HabbajetScreen />
        </Provider>
    ),
    () => HabbajetScreen,
);

export const goToLoading = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Loading,
                            options: {
                                topBar: {
                                    visible: false,
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};

export const goToHabbajet = () => {
    Navigation.setRoot({
        root: {
            stack: {
                id: STACK_NAVIGATOR,
                children: [
                    {
                        component: {
                            name: Screens.Habbajet,
                            options: {
                                topBar: {
                                    title: {
                                        text: ' ',
                                        fontFamily: 'Abel',
                                        fontSize: 30,
                                        color: '#ffffff',
                                    },
                                },
                            },
                        },
                    },
                ],
            },
        },
    });
};
