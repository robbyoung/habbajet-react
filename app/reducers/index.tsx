import habbajets from './habbajets';
import purchases from './purchases';
import budget from './budget';
import habbajetEditor from './habbajetEditor';
import purchaseEditor from './purchaseEditor';
import tagEditor from './tagEditor';
import tags from './tags';
import {combineReducers} from 'redux';

export default combineReducers({
    habbajets,
    purchases,
    budget,
    tags,
    habbajetEditor,
    purchaseEditor,
    tagEditor,
});
