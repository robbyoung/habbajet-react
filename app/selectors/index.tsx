export * from './budget';
export * from './purchases';
export * from './habbajets';
export * from './habbajetEditor';
