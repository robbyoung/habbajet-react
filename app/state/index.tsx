export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
