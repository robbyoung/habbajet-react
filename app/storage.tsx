import AsyncStorage from '@react-native-community/async-storage';
import {State} from './state';
import store from './store';
import {createTestState} from './state/testState';

// const defaultState: State = {
//     budget: 0,
//     habbajets: [],
//     purchases: [],
// };

const defaultState = createTestState(1, 0, 0);

export async function saveState(): Promise<void> {
    const state = store.getState();
    await AsyncStorage.setItem('state', JSON.stringify(state));
}

export async function loadState(): Promise<State> {
    const state = await AsyncStorage.getItem('state');
    let savedState: State;
    if (state !== undefined && state !== null) {
        savedState = JSON.parse(state) as State;
        return savedState;
    }
    return defaultState;
}

export async function clearState() {
    await AsyncStorage.clear();
}
