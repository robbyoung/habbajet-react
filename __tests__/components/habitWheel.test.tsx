import React from 'react';
import renderer from 'react-test-renderer';
import HabitWheel from '../../app/components/habitWheel';
import {habbajetColors} from '../../app/colors';

describe('HabitWheel Component', () => {
    it('renders an empty circle with no successes', () => {
        const component = renderer.create(
            <HabitWheel color={habbajetColors[0]} successes={0} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a half-full circle with some successes', () => {
        const component = renderer.create(
            <HabitWheel color={habbajetColors[1]} successes={4} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a full circle with all successes', () => {
        const component = renderer.create(
            <HabitWheel color={habbajetColors[2]} successes={7} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('handles invalid success counts', () => {
        const component = renderer.create(
            <HabitWheel color={habbajetColors[3]} successes={-8} />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
