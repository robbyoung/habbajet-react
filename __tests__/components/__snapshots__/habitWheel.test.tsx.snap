// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitWheel Component handles invalid success counts 1`] = `
<View
  style={
    Object {
      "margin": 30,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 690,
          "width": 690,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          6.123233995736766e-17,
          -1,
          1,
          6.123233995736766e-17,
          0,
          690,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            517.5,
            345,
            4,
            345,
            345,
            172.5,
            0,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#f5f5f5"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTGroup
        opacity={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`HabitWheel Component renders a full circle with all successes 1`] = `
<View
  style={
    Object {
      "margin": 30,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 690,
          "width": 690,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          6.123233995736766e-17,
          -1,
          1,
          6.123233995736766e-17,
          0,
          690,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            517.5,
            345,
            4,
            345,
            345,
            172.5,
            0,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#f5f5f5"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            517.394917660794,
            351.02016318118143,
            4,
            345,
            345,
            172.5,
            0.03490658503988659,
            0.8984954989266809,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e0474721"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            447.6553277862689,
            483.62948343369703,
            4,
            345,
            345,
            172.5,
            0.9334020839665675,
            1.7969909978533618,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e0474742"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            300.4700498337862,
            511.65333341459007,
            4,
            345,
            345,
            172.5,
            1.8318975828932482,
            2.6954864967800427,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e0474763"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            186.8792545552059,
            413.94983582273875,
            4,
            345,
            345,
            172.5,
            2.73039308181993,
            3.593981995706724,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e0474785"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            192.5786139568356,
            264.2288970195365,
            4,
            345,
            345,
            172.5,
            3.6288885807466102,
            4.4924774946334045,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e04747a6"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            313.26839201930574,
            175.4436522717019,
            4,
            345,
            345,
            172.5,
            4.527384079673291,
            5.390972993560085,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e04747c7"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            457.89727063193357,
            214.5758600417087,
            4,
            345,
            345,
            172.5,
            5.425879578599972,
            6.289468492486765,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#e04747e8"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTGroup
        opacity={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`HabitWheel Component renders a half-full circle with some successes 1`] = `
<View
  style={
    Object {
      "margin": 30,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 690,
          "width": 690,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          6.123233995736766e-17,
          -1,
          1,
          6.123233995736766e-17,
          0,
          690,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            517.5,
            345,
            4,
            345,
            345,
            172.5,
            0,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#f5f5f5"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            517.394917660794,
            351.02016318118143,
            4,
            345,
            345,
            172.5,
            0.03490658503988659,
            0.8984954989266809,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#2aad5e21"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            447.6553277862689,
            483.62948343369703,
            4,
            345,
            345,
            172.5,
            0.9334020839665675,
            1.7969909978533618,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#2aad5e42"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            300.4700498337862,
            511.65333341459007,
            4,
            345,
            345,
            172.5,
            1.8318975828932482,
            2.6954864967800427,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#2aad5e63"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            186.8792545552059,
            413.94983582273875,
            4,
            345,
            345,
            172.5,
            2.73039308181993,
            3.593981995706724,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#2aad5e85"
        strokeCap={0}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTGroup
        opacity={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;

exports[`HabitWheel Component renders an empty circle with no successes 1`] = `
<View
  style={
    Object {
      "margin": 30,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <ARTSurfaceView
    style={
      Array [
        undefined,
        Object {
          "height": 690,
          "width": 690,
        },
      ]
    }
  >
    <ARTGroup
      opacity={1}
      transform={
        Array [
          6.123233995736766e-17,
          -1,
          1,
          6.123233995736766e-17,
          0,
          690,
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            517.5,
            345,
            4,
            345,
            345,
            172.5,
            0,
            6.283185307179586,
            1,
          ]
        }
        fill={null}
        opacity={1}
        stroke="#f5f5f5"
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={345}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
      <ARTGroup
        opacity={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      />
    </ARTGroup>
  </ARTSurfaceView>
</View>
`;
